File: Daily Challanges/2026-04-April/2026-04-03.js
function getBrowserHistory(commands) {
  let history = [];
  let current = -1;

  for (let cmd of commands) {
    if (cmd === "Back") {
      if (current > 0) {
        current--;
      }
    } else if (cmd === "Forward") {
      if (current < history.length - 1) {
        current++;
      }
    } else {
      history = history.slice(0, current + 1);
      history.push(cmd);
      current++;
    }
  }

  return [history, current];
}

File: Daily Challanges/2026-04-April/2026-04-03.py
def get_browser_history(commands):
    history = []
    current = -1
    for cmd in commands:
        if cmd == "Back":
            if current > 0:
                current -= 1
        elif cmd == "Forward":
            if current < len(history) - 1:
                current += 1
        else:  
            history = history[:current + 1]
            history.append(cmd)
            current += 1
    
    return [history, current]